orion/sites/admin/src/home/home.js

```diff
@@ -2,30 +2,211 @@
+            switch: true,
+        });
+        if (programs.length == 0) {
+            this.setState({
+                noUnpub: "All items are published! Your website is up to date!",
+                switch: false,
+            });
+        }
+    }
+
+    onClickClasses() {
+        let classes = this.state.classes.map((row, index) => {
+            return <DashboardRow key={index} title={row.classId} />;
+        });
+        this.setState({
+            noUnpub: "",
+            rows: classes,
+            selection: "classes",
+            switch: true,
+        });
+        if (classes.length == 0) {
+            this.setState({
+                noUnpub: "All items are published! Your website is up to date!",
+                switch: false,
+            });
+        }
+    }
+
+    onClickLocations() {
+        let locations = this.state.locations.map((row, index) => {
+            return <DashboardRow key={index} title={row.locId} />;
+        });
+        this.setState({
+            noUnpub: "",
+            rows: locations,
+            selection: "locations",
+            switch: true,
+        });
+        if (locations.length == 0) {
+            this.setState({
+                noUnpub: "All items are published! Your website is up to date!",
+                switch: false,
+            });
+        }
+    }
+
+    onClickAchievements() {
+        let achieves = this.state.achieves.map((row, index) => {
+            return <DashboardRow key={index} title={row.message} />;
+        });
+        this.setState({
+            noUnpub: "",
+            rows: achieves,
+            selection: "achievements",
+            switch: true,
+        });
+        if (achieves.length == 0) {
+            this.setState({
+                noUnpub: "All items are published! Your website is up to date!",
+                switch: false,
+            });
+        }
+    }
+
+    onClickSemesters() {
+        let semesters = this.state.semesters.map((row, index) => {
+            return <DashboardRow key={index} title={row.semesterId} />;
+        });
+        this.setState({
+            noUnpub: "",
+            rows: semesters,
+            selection: "semesters",
+            switch: true,
+        });
+        if (semesters.length == 0) {
+            this.setState({
+                noUnpub: "All items are published! Your website is up to date!",
+                switch: false,
+            });
+        }
+    }
+
+    onClickSessions() {
+        let sessions = this.state.sessions.map((row, index) => {
+            return <DashboardRow key={index} title={row.sessionId} />;
+        });
+        this.setState({
+            noUnpub: "",
+            rows: sessions,
+            selection: "sessions",
+            switch: true,
+        });
+        if (sessions.length == 0) {
+            this.setState({
+                noUnpub: "All items are published! Your website is up to date!",
+                switch: false,
+            });
+        }
```

Comment:
This looks so much neater. Here's one more thing you could do:
Ideally, it would be great to have only ONE this.setState(...). And you can do so by using this fancy one-line if else notation.

It works like this:
`<CONDITION> ? <result if true> : <result if false>`
This is also a thing in Java too actually and a few other languages. It's just an easy way to do a one-line if/else statement and you can assign the result to a variable.

So your code can be like this:
```
let noUnpub = sessions.length == 0 ? "..." : "";
let switch = sessions.length == 0 ? false : true
```

And now you can use one this.setState(...) after these two statements.

Reply:
wow this fancy if else notation is pretty nice! I put them inside the setState still though.